include/deal.II/lac/tensor_product_matrix.h

```diff
@@ -382,10 +402,9 @@ class TensorProductMatrixSymmetricSumCollection
    * stiffness matrices @p Ks to the stored data, looking out for
    * compression possibilities.
    */
+  template <typename T>
   void
-  insert(const unsigned int                       index,
-         const std::array<Table<2, Number>, dim> &Ms,
-         const std::array<Table<2, Number>, dim> &Ks);
+  insert(const unsigned int index, const T &Ms, const T &Ks);
```

Comment:
- generalize initialization just like in the case of `TensorProductMatrixSymmetricSum`